### ISSUE TITLE
standard library tests are failing when using the C backend 

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#15330]
Zig Version: `0.11.0-dev.2634+b0186f310`

https://github.com/ziglang/zig/blob/b0186f31002760c9940d8c06c2ac7baa9d52f404/test/tests.zig#L970-L975

Until this issue is resolved, the CI does not provide test coverage for the C backend passing the std lib tests.

**Comment by jacobly0** [comment]
After #15356, this issue will track the remaining std tests disabled with the C backend due to missing features.  This can be closed when there are no more std tests disabled for the C backend that are not already tracked by another issue.
